packages/@internationalized/string/src/LocalizedStringDictionary.ts

```diff
@@ -30,7 +30,10 @@ export class LocalizedStringDictionary<K extends string = string, T extends Loca
 
   constructor(messages: LocalizedStrings<K, T>, defaultLocale: string = 'en-US') {
     // Clone messages so we don't modify the original object.
-    this.strings = {...messages};
+    // Filter out entries with falsy values which may have been caused by applying optimize-locales-plugin.
+    this.strings = Object.fromEntries(
```

Comment:
Did you test if this was actually necessary? The logic below in getStringsForLocale looks like it would work with undefined values already.

Reply:
Yes I did. One of tests I added is failing with old code and passing with my changes.

I intentionally didn't touch code in getStringsForLocale because there was mentioned there are plans to replace it with standard Intl.LocaleMatcher.